optuna/trial/_frozen.py

```diff
@@ -166,29 +170,72 @@ def _validate(self):
                 )
 
     @property
-    def distributions(self):
-        # type: () -> Dict[str, BaseDistribution]
+    def number(self) -> int:
+        return self._number
+
+    @number.setter
+    def number(self, value: int) -> None:
+        self._number = value
+
+    @property
+    def datetime_start(self) -> Optional[datetime.datetime]:
+        return self._datetime_start
+
+    @datetime_start.setter
+    def datetime_start(self, value: Optional[datetime.datetime]) -> None:
+        self._datetime_start = value
+
+    def set_user_attr(self, key: str, value: Any) -> None:
+
+        self._user_attrs[key] = value
+
+    def set_system_attr(self, key: str, value: Any) -> None:
+
+        self._system_attrs[key] = value
```

Comment:
It might be helpful for code readers to move these lines after `should_prune`. The order of the functions should match with that of `optuna/trial/_trial.py`.

Reply:
Absolutely!

By the way, I found that In `fixedTrial.py` and `Trial.py`, `_suggest`'s position is not consistent:

https://github.com/optuna/optuna/blob/0ec8ddc1809f7d3f315b119c51a8090452364a0a/optuna/trial/_fixed.py#L130
`_suggest` function appears just after public `suggest_*`, then other public functions are defined after `_suggest`.

https://github.com/optuna/optuna/blob/0ec8ddc1809f7d3f315b119c51a8090452364a0a/optuna/trial/_trial.py#L660
`_suggest` function appears after all public function.

Reply:
Thanks for the good catch! I think it is better to align the `optuna/trial/_trial.py`. (The `optuna/trial/_fixed.py` should be fixed in another PR.)

Reply:
I created the PR :)
https://github.com/optuna/optuna/pull/1528